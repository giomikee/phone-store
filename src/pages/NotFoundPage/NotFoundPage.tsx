import { Box, Button, Typography } from '@mui/material';
import ReportIcon from '@mui/icons-material/Report';
import UndoIcon from '@mui/icons-material/Undo';
import { Link } from 'react-router-dom';
import { PAGES } from '../../constants';

const NotFoundPage = () => {
    return (
        <>
            <ReportIcon 
                color='error'
                sx={{
                    height: '150px',
                    width: '100%'
                }}
            />

            <Box sx={{ flexGrow: 1, textAlign: 'center'}}>
                <Typography 
                    component='h2' 
                    gutterBottom
                    variant='h5'
                >
                    PÃ¡gina no encontrada.
                </Typography>

                <Link to={PAGES.home}>
                    <Button
                        startIcon={<UndoIcon />}
                        variant='outlined'
                    >
                        Volver al inicio
                    </Button>
                </Link>
            </Box>
        </>
    )
}

export default NotFoundPage