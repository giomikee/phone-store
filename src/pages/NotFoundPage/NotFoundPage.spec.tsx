import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import { BrowserRouter } from 'react-router-dom';
import NotFoundPage from './NotFoundPage';

describe('Given NotFoundPage component', () => {
    describe('When the component is rendered', () => {
        it('Then it should match the snapshot', () => {
            const { container } = render(<NotFoundPage />, { wrapper: BrowserRouter });
            expect(container).toMatchSnapshot();
        });
    });
});