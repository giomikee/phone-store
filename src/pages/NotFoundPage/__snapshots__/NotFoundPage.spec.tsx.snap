// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given NotFoundPage component > When the component is rendered > Then it should match the snapshot 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-q2ex3j-MuiSvgIcon-root"
    data-testid="ReportIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M15.73 3H8.27L3 8.27v7.46L8.27 21h7.46L21 15.73V8.27zM12 17.3c-.72 0-1.3-.58-1.3-1.3s.58-1.3 1.3-1.3 1.3.58 1.3 1.3-.58 1.3-1.3 1.3m1-4.3h-2V7h2z"
    />
  </svg>
  <div
    class="MuiBox-root css-17ls7xt"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1f9xtph-MuiTypography-root"
    >
      PÃ¡gina no encontrada.
    </h2>
    <a
      href="/"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1blvszr-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="UndoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8"
            />
          </svg>
        </span>
        Volver al inicio
      </button>
    </a>
  </div>
</div>
`;
